crates/sui-types/src/programmable_transaction_builder.rs

```diff
@@ -168,16 +168,26 @@ impl ProgrammableTransactionBuilder {
         Ok(())
     }
 
-    pub fn publish(&mut self, modules: Vec<Vec<u8>>) {
+    pub fn publish_and_handle_capability_with(
+        &mut self,
+        modules: Vec<Vec<u8>>,
+        cap_handler: impl FnOnce(&mut ProgrammableTransactionBuilder, Argument) -> Command,
+    ) {
```

Comment:
If we make the other `publish_immutable`. I think it might just be simpler to make `publish` return an argument, e.g. `pub fn publish(&mut self, modules: Vec<Vec<u8>>) -> Argument`. I think that feels a bit more in line with the other functions here. Thoughts?

Reply:
+1, probably better to return the `Argument` here.

Reply:
Sounds good! I was a bit on the fence about whether it should return the `Argument` or take the callback, so will make this change :)